types/hls.js/index.d.ts

```diff
@@ -1366,7 +1432,7 @@ declare namespace Hls {
     /**
      * levelDetails object
      */
-    details?: LevelDetails;
+    details?: string;
```

Comment:
Yes this was intentional to match API doc https://github.com/video-dev/hls.js/blob/master/doc/API.md#fifth-step-error-handling

I will make a quick commit to update the docs.